pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -77,8 +75,22 @@ func GetAPIClient() (*APIClient, error) {
 	return globalAPIClient, nil
 }
 
-// getClient returns an official Kubernetes core v1 client
-func getClient() (*corev1.CoreV1Client, error) {
+// getClientSet returns the generic kubernetes client set
+func getClientSet() (*kubernetes.Clientset, error) {
+	k8sConfig, err := getK8sConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientSet, err := kubernetes.NewForConfig(k8sConfig)
```

Comment:
we're only using `core` and `autoscaling` client is it worth using the client set or should we only use the two clients?